File: Day-13/339A.Helpful Maths.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main() {
    string s;
    cin >> s;
    vector<int> numbers;
    
    for (char c : s) {
        if (c != '+') {
            numbers.push_back(c - '0');
        }
    }
    
    sort(numbers.begin(), numbers.end());
    
    for (int i = 0; i < numbers.size(); ++i) {
        if (i != 0) {
            cout << "+";
        }
        cout << numbers[i];
    }
    
    return 0;
}

File: Day-13/339A.Helpful Maths.java
import java.util.Arrays;
import java.util.Scanner;

public class HelpfulMaths {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String s = scanner.nextLine();
        String[] parts = s.split("\\+");
        int[] numbers = new int[parts.length];
        
        for (int i = 0; i < parts.length; i++) {
            numbers[i] = Integer.parseInt(parts[i]);
        }
        
        Arrays.sort(numbers);
        
        StringBuilder result = new StringBuilder();
        for (int i = 0; i < numbers.length; i++) {
            if (i != 0) {
                result.append("+");
            }
            result.append(numbers[i]);
        }
        
        System.out.println(result.toString());
    }
}
